parsl/data_provider/data_manager.py

```diff
@@ -140,8 +142,12 @@ def stage_in(self, file, executor):
     def _file_stage_in_app(self):
         return App("python", executors=['data_manager'])(self._file_stage_in)
 
-    def _file_stage_in(self, outputs=[]):
-        pass
+    def _file_stage_in(self, executor_label, outputs=[]):
+        file = outputs[0]
+        executor = self.executors[executor_label]
+        if isinstance(executor, IPyParallelExecutor):
+            executor.provider.channel.push_file(file.filepath, executor.working_dir)
```

Comment:
@annawoodard I'm a bit confused here. This would only work for sites where the login node and compute nodes have a shared file system. Am I missing something obvious ?